app/src/main/java/dev/andrylat/raqimbek/bankingutils/BankingUtilsApp.java

```diff
@@ -15,29 +14,63 @@ public static void main(String[] args) {
   }
 
   private static void run() {
-    userInteraction.write(false, List.of("Hello. Enter card number for validation:"));
+    StringBuilder greetingMessage =
+        new StringBuilder("Hello. Please type the index of the service you need:\n");
+    var services = BankingService.values();
 
-    var userInput = userInteraction.read();
-    var cardValidationInfo = cardValidator.checkCardNumber(userInput);
+    for (var i = 0; i < services.length; i++) {
+      greetingMessage.append("[").append(i).append("] - ").append(services[i]).append("\n");
+    }
+
+    commandLineUserInteraction.write(greetingMessage.toString());
+
+    var selectedService = 0;
+
+    try {
+      selectedService = Integer.parseInt(commandLineUserInteraction.read());
+    } catch (NumberFormatException e) {
+      commandLineUserInteraction.write(
+          "Please write only an integer number representing an index of a service.");
+    }
+
+    switch (services[selectedService]) {
+      case BankingService.CARD_VALIDATOR:
+        commandLineUserInteraction.write(BankingService.CARD_VALIDATOR.getGreetingMessage());
+
+        var userInput = commandLineUserInteraction.read();
+        var cardValidationInfo = cardValidator.checkCardNumber(userInput);
+
+        if (!cardValidationInfo.isValid()) {
+          var errors = cardValidationInfo.errors();
+
+          commandLineUserInteraction.write("Card is not valid. Errors:");
+          commandLineUserInteraction.writeAll(errors);
+        } else {
+          var paymentSystemOptional =
+              paymentSystemDeterminer.determinePaymentSystemByCardNumber(userInput);
 
-    if (!cardValidationInfo.getIsValid()) {
-      var errors = cardValidationInfo.getErrors();
+          if (paymentSystemOptional.isPresent()) {
+            var paymentSystem = paymentSystemOptional.get();
+            var message =
+                new StringBuilder("Card is valid. Payment System is \"")
+                    .append(paymentSystem)
+                    .append("\"")
+                    .toString();
 
-      userInteraction.write(true, errors);
-    } else {
-      var paymentSystemOptional =
-          paymentSystemDeterminer.determinePaymentSystemByCardNumber(userInput);
+            commandLineUserInteraction.write(message);
+          }
+        }
+        break;
+      case BankingService.MORTGAGE_CALCULATOR:
```

Comment:
Tomorrow I will add another feature. And day after tomorrow - one more feature. This code will grow to infinity? 

use interfaces and implementations. Use interface like "Dialog" and create 2 implementations "CardValidatorDialog" and "MortgageCalculatorDialog". Depends on user's options, you will run necessary dialog.

Also you can create map where key is option and value is dialog implementation to run.


finally code should be "dialogs.get(usersOption).run()".

And when new feature will come, you will need to create new dialog implementation and add new value to the map ONLY.